### PR TITLE
add `defaultValue` @ `multioption`.

### PR DESCRIPTION
**Pull request by igalklebanov**
Hey :wave:

This PR adds the ability to provide a default value array in case the user didn't provide any options to a multioption field.

e.g. no entrypoints provided, assume `['./migrations']`.

Currently, there's no `defaultValue` configuration, and it always falls back to `[]`. It'll still fallback to `[]` after these changes if developer didn't provide `defaultValue`.

### CODE REVIEW

**Review by naorpeled — APPROVED** [review]

File: test/multioption.test.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+});
+
+test('fails when no option is provided and applying default value fails', async () => {
+  const argv = '';
+  const tokens = tokenize(argv.split(' '));
+  const argparser = multioption({
+    type: array(string),
+    long: 'hello',
+    defaultValue: () => {
+      throw new Error('its too hot outside, stay inside muthafucka!');
```
Comment (Schniz) [inline test/multioption.test.ts:79]: Please don't use swear words in the codebase <picture data-single-emoji=":blob-angel:" title=":blob-angel:">:blob-angel:</picture>

File: src/multioption.ts [hunk 2]
```diff
@@ -53,6 +75,24 @@ export function multioption<Decoder extends Type<string[], any>>(
         shortNames: config.short ? [config.short] : [],
       }).filter((x) => !visitedNodes.has(x));
 
+      const defaultValueFn = config.defaultValue ?? config.type.defaultValue;
+
+      if (options.length === 0 && typeof defaultValueFn === 'function') {
+        try {
+          return Result.ok(defaultValueFn());
+        } catch (e: any) {
+          const message = `Default value not found for '--${config.long}': ${e.message}`;
```
Comment (Schniz) [inline src/multioption.ts:84]:
```suggestion
          const message = `Failed to resolve default value for '--${config.long}': ${e.message}`;
```

**Review by Schniz — APPROVED** [review]
<picture data-single-emoji=":lfgtm:" title=":lfgtm:">:lfgtm:</picture> <picture data-single-emoji=":lfg-intense:" title=":lfg-intense:">:lfg-intense:</picture> 

Thanks for the hard work <picture data-single-emoji=":galactic_relifed:" title=":galactic_relifed:">:galactic_relifed:</picture> 

Let's fix the swear words and the error message and ship this <picture data-single-emoji=":shipped:" title=":shipped:">:shipped:</picture>

File: test/multioption.test.ts (hunk 1 above)
Comment (igalklebanov) [reply test/multioption.test.ts]: ok.. 😞

**Comment by Schniz** [comment]
Can you please fix the assertions? 🙏

**Comment by Schniz** [comment]
Wooohooooo

**Pull request merged**